components/playlist/resources/components/playlistSelect.tsx

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+
+import * as PlaylistMojo from 'gen/brave/components/playlist/mojom/playlist.mojom.m.js'
+
+interface Props {
+  playlists: PlaylistMojo.Playlist[]
+  selectedPlaylist: PlaylistMojo.Playlist | undefined
+
+  selectPlaylist: (playlistId: string) => void
+  createPlaylist: (playlist: PlaylistMojo.Playlist) => void
+  removePlaylist: (playlistId: string) => void
+};
+
+export default function PlaylistSelect ({
+    playlists,
+    selectedPlaylist,
+    selectPlaylist,
+    createPlaylist,
+    removePlaylist
+}: Props) {
+   const onClickCreateButton = React.useCallback((e) => {
```

Comment:
So I normally use `useCallback` too but Pete actually recommends not using it, unless the component is in a `useMemo` and causes a lot of different things to rerender, as it's a pretty minor optimisation.

This also means you can directly inline the simple event handlers! (if you like, otherwise you can just leave them here without the `useCallback`)

```tsx

<select onChange={e => selectPlaylist(e.target.value)}>
...
</select>
```

The more complicated ones are probably better to leave :D

Reply:
You can remove the other `useCallbacks` as well if you want, but not too important.